Blog/tests.py

```diff
@@ -52,6 +53,27 @@ def test_verify_edit_link_correct_on_first_fetch_incorrect_on_sec_fetch(self):
         self.assertNotContains(content_response, data, status_code=200)
 
 
+class EditLinkTest(TestCase):
+    def setUp(self):
+        self.blogpost = Blog.objects.create(title='test blog post with ORM', content='test content by a post with ORM')
+        self.blogpost.secret_key = uuid.uuid4().hex[:6].upper()
+        self.blogpost.save()
+        self.instance = Blog.objects.get(title='test blog post with ORM')
+
+    def test_verify_redirection_of_edit_link(self):
+        response = self.client.get(reverse('Blog:homepage', kwargs={'id': self.instance.id, 'secret_key': self.instance.secret_key}))
+        self.assertContains(response, 'test blog post with ORM', status_code=200)
+        wrong_secret_key = 'DF20192'
+        response = self.client.get(reverse('Blog:homepage', kwargs={'id': self.instance.id, 'secret_key': wrong_secret_key}))
+        self.assertNotContains(response, 'test blog post with ORM', status_code=404)
+
+    def test_changes_edits_the_post_successfully(self):
+        update_data = {'title': 'test title to check the updation', 'content': 'test content to check the updation is working'}
+        response = self.client.post(reverse('Blog:homepage', kwargs={'id': self.instance.id, 'secret_key': self.instance.secret_key}), update_data, follow=True)
+        self.assertContains(response, 'Your post has been successfully Updated!!!', status_code=200)
```

Comment:
To test if an update happened, fetch the blog again and then check the content. Dont rely on anything other than that.